src/Features/CSharp/Portable/ValidateFormatString/CSharpValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+            }
+
+            argsArgumentType = semanticModel.GetTypeInfo(argsArgument.Expression).Type;
+            return true;
+        }
+
+        private static bool TryGetArgument(
+            SemanticModel semanticModel,
+            string searchArgumentName,
+            SeparatedSyntaxList<ArgumentSyntax> arguments,
+            ImmutableArray<IParameterSymbol> parameters,
+            out ArgumentSyntax argumentSyntax)
+        {
+            argumentSyntax = null;
+
+            // First, look for a named argument that matches
+            var namedArgument = arguments.SingleOrDefault(
+                p => p.NameColon?.Name.Identifier.ValueText.Equals(searchArgumentName) == true);
+
+            if (namedArgument != null)
+            {
+                argumentSyntax = namedArgument;
+                return true;
+            }
+
+            // If no named argument exists, look for the named parameter
+            // and return the corresponding argument
+            var namedParameter = parameters.SingleOrDefault(p => p.Name.Equals(searchArgumentName) == true);
+            if (namedParameter != null)
+            {
+                // For the case string.Format("Test string"), there is only one argument
+                // but the compiler created an empty parameter array to bind to an overload
+                if (namedParameter.Ordinal >= arguments.Count)
+                {
+                    return false;
+                }
+
+                // Multiple arguments could have been converted to a single params array, 
+                // so there wouldn't be a corresponding argument
+                if (namedParameter.IsParams && parameters.Length != arguments.Count)
+                {
+                    return false;
+                }
+
+                argumentSyntax = arguments[namedParameter.Ordinal];
+                return true;
+            }
+
+            return false;
+        }
+
+        protected override bool TryGetFormatStringLiteralExpressionSyntax(
+            SemanticModel semanticModel,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters,
+            out SyntaxNode formatStringLiteralExpressionSyntax)
+        {
+            formatStringLiteralExpressionSyntax = null;
+
+            if (!TryGetArgument(semanticModel, "format", arguments, parameters, out var formatArgumentSyntax))
+            {
+                return false;
+            }
+
+            if (!formatArgumentSyntax.Expression.IsKind(SyntaxKind.StringLiteralExpression))
+            {
+                return false;
+            }
+
+            formatStringLiteralExpressionSyntax = formatArgumentSyntax.Expression;
+            return true;
+        }
+
+        protected override SyntaxKind GetInvocationExpressionSyntaxKind()
+            => SyntaxKind.InvocationExpression;
+
+        protected override string GetLiteralExpressionSyntaxAsString(SyntaxNode syntaxNode) 
```

Comment:
This does not special language knowledge.  You can just call .ToString on the node at the base layer, right?